src/bun.js/node/node_fs_watcher.zig

```diff
@@ -603,12 +608,15 @@ pub const FSWatcher = struct {
     }
 
     pub fn emitAbort(this: *FSWatcher, err: JSC.JSValue) void {
-        if (this.isClosed()) return;
+        if (this.closed) return;
         defer this.close(true);
 
         err.ensureStillAlive();
         if (this.js_this != .zero) {
-            if (FSWatcher.listenerGetCached(this.js_this)) |listener| {
+            const js_this = this.js_this;
+            js_this.ensureStillAlive();
+            if (FSWatcher.listenerGetCached(js_this)) |listener| {
```

Comment:
what happens if the user calls `.close()` in the callback and there are no file changes?

Reply:
the user will call close and its already closed when the code call close again, and the unref will only happen after both all on main thread.